tests/test_core.py

```diff
@@ -391,17 +392,38 @@ def test_read_catalog_dict():
             {'A': 'NCAR', 'B': 'CESM', 'C': 'control', 'D': 'O2'},
         ],
     ),
+    (
+        {'D': [re.compile(r'^O2$'), 'NO2'], 'B': ['CESM', 'BAR']},
+        None,
+        [
+            {'A': 'NCAR', 'B': 'CESM', 'C': 'hist', 'D': 'O2'},
+            {'A': 'CSIRO', 'B': 'BAR', 'C': 'control', 'D': 'O2'},
+            {'A': 'NCAR', 'B': 'CESM', 'C': 'control', 'D': 'O2'},
+        ],
+    ),
+    ({'C': ['hist'], 'D': ['TA']}, None, [{'A': 'NCAR', 'B': 'WACM', 'C': 'hist', 'D': 'TA'}],),
+    (
+        {
+            'C': [re.compile('hist.*', flags=re.IGNORECASE)],
+            'D': [re.compile('TA.*', flags=re.IGNORECASE)],
```

Comment:
@sherimickelson, I was able to test that the search function is respecting regex flags such as case sensitivity.